docs/Z-Score-normalization-script.md

```diff
@@ -5,31 +5,54 @@ For some data types, when uploading to cBioPortal, it is currently necessary to
+Each **gene** is normalized separately. 
+
+We currently have two profiles for generating z-scores using two different base population:
+
+- Distribution based on **diploid** samples only: The expression distribution for unaltered copies of the gene is estimated by calculating the mean and variance of the expression values for samples in which the gene is diploid (i.e. value is "0" as reported by [discrete CNA data](File-Formats.md#discrete-copy-number-data)). We call this the unaltered distribution. If the gene has no diploid samples, then its normalized expression is reported as NA. 
+
+- Distribution based on **all** samples: The expression distribution of the gene
+is estimated by calculating the mean and variance of the expression values for samples whose values are not Null, NA or NaN. If the gene has samples whose expression values are Null, NaN or 'NA', then its normalized expression is reported as NA.
+
+Otherwise when valid expression values exist, with both profiles, for every sample, the gene's normalized expression is reported as
 
 ```
 (r - mu)/sigma
 ```
 
-where `r` is the raw expression value, and `mu` and `sigma` are the mean and standard deviation of the unaltered distribution, respectively.
+where `r` is the raw expression value, and `mu` and `sigma` are the mean and standard deviation of the base population, respectively.
+
+### How to proceed
+cBioPortal expects z-score normalization to take place per gene. You can produce this extra z-score file yourself, or let cBioPortal do it for your input files by using `convertExpressionZscores.pl` (Method 1, diploid samples as base population) OR `NormalizeExpressionLevels.py` (Method 2, all samples as base population). Examples of the calculation and running the programs are below.
 
 ## Parameters
 
+### `convertExpressionZscores.pl`
+
 `<copy_number_file> <expression_file> <output_file> <normal_sample_suffix> <[min_number_of_diploids]>`
 
 - `<copy_number_file>` : the [discrete copy number (CNA) file](File-Formats.md#discrete-copy-number-data) 
 - `<expression_file>` : the [expression (exp) data file](File-Formats.md#expression-data). 
 - `<output_file>` : the output file to be generated
 - `<normal_sample_suffix>` : use this to identify which of your samples are "normal" samples (if any). E.g. normal TCGA samples have a suffix "-11". Set it to some dummy value, e.g. "NONE", if you have no normal samples in your data.
 
-## convertExpressionZscores Transformation Algorithm 
+### `NormalizeExpressionLevels.py`
+
+`-i <expression_file> -o <output_file>`
+
+- `<expression_file>` : the [expression (exp) data file](File-Formats.md#expression-data).
+- `<output_file>` : the output file to be generated
+- We require the mRNA expression data to be log transformed before calculating zscores on them. If the data is not already log tranformed use the `-l` option to calculate log base 2 of the expression values.
+
+## Transformation Algorithm 
```

Comment:
This header confused me at first. Maybe call it something like "Algorithms & Implementation"? Also you have both algorithm & implementation sections for convertExpressionZscores but only one for NormalizeExpressionLevels

Reply:
Merged algorithm and implementation sections.